io-engine/src/rebuild/rebuild_log.rs

```diff
@@ -0,0 +1,158 @@
+use bit_vec::BitVec;
+use parking_lot::Mutex;
+use spdk_rs::IoType;
+use std::{
+    fmt::{Debug, Formatter},
+    sync::Arc,
+};
+
+use super::SEGMENT_SIZE;
+
+// Returns ceil of an integer division.
+fn div_ceil(a: u64, b: u64) -> u64 {
+    (a + b - 1) / b
+}
+
+/// Rebuild log data structure.
+pub struct RebuildLog {
+    /// Name of the underlying device, stored for diagnostic purposes.
+    device_name: String,
+    /// Segment bitmap. Ones indicate modified segments.
+    segments: BitVec,
+    /// Device size in segments.
+    num_segments: u64,
+    /// Device size in block.
+    num_blocks: u64,
+    /// Size of block in bytes.
+    block_len: u64,
+}
+
+impl Debug for RebuildLog {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Rebuild log: '{}' ({} segments: {} blocks x {})",
+            self.device_name,
+            self.num_segments,
+            self.num_blocks,
+            self.block_len
+        )
+    }
+}
+
+impl RebuildLog {
+    /// Creates a new rebuild log instance for the given device.
+    pub fn new(device_name: &str, num_blocks: u64, block_len: u64) -> Self {
+        let num_segments = div_ceil(num_blocks * block_len, SEGMENT_SIZE);
+        let mut segments = BitVec::new();
+        segments.grow(num_segments as usize, false);
+        Self {
+            device_name: device_name.to_owned(),
+            segments,
+            num_segments,
+            num_blocks,
+            block_len,
+        }
+    }
+
+    /// Logs the given operation, marking the corresponding segment as modified
+    /// in the case of a write operation.
+    ///
+    /// # Arguments
+    ///
+    /// * `_io_type`: IoType of the operation to log.
+    /// * `lbn`: Logical block number.
+    /// * `lbn_cnt`: Number of logical blocks affected by the operation.
+    fn log_op(&mut self, _io_type: IoType, lbn: u64, lbn_cnt: u64) {
```

Comment:
Do we need `_io_type`, eg: we only need to set bit for write-like operations?

Reply:
Currently, it's here for debugging purposes.